src/js/game/systems/display.js

```diff
@@ -94,4 +94,50 @@ export class DisplaySystem extends GameSystemWithFilter {
             }
         }
     }
+
+    /**
+     * Draws overlay of a given chunk
+     * @param {import("../../core/draw_utils").DrawParameters} parameters
+     * @param {MapChunkView} chunk
+     */
+    drawChunkOverlay(parameters, chunk) {
+        const contents = chunk.containedEntitiesByLayer.regular;
+        for (let i = 0; i < contents.length; ++i) {
+            const entity = contents[i];
+            if (!entity || !entity.components.Display) {
+                continue;
+            }
+
+            const pinsComp = entity.components.WiredPins;
+            const network = pinsComp.slots[0].linkedNetwork;
+
+            if (!network || !network.currentValue) {
+                continue;
+            }
+
+            const value = this.getDisplayItem(network.currentValue);
+
+            if (!value) {
+                continue;
+            }
+
+            const origin = entity.components.StaticMapEntity.origin;
+            if (value instanceof ColorItem) {
+                this.displaySprites[value.color].drawCachedCentered(
+                    parameters,
+                    (origin.x + 0.5) * globalConfig.tileSize,
+                    (origin.y + 0.5) * globalConfig.tileSize,
+                    globalConfig.tileSize
+                );
+
+                parameters.context.fillStyle = enumColorsToHexCode[value.color];

```

Comment:
Why are you drawing the display sprite here as well as fillRect?

Reply:
lol, weird copy paste i guess